### PR TITLE
[ci skip] Active Record Validations guide fixes

### PR DESCRIPTION
**Pull request by jonatack**
- Improve the text explanations for the #add and #full_messages methods
- Remove deprecated `[]=` setter method.
- Fix the #add method that was shown in duplicate. The second code example was originally `[]=`. Replace it with `[] <<`.

### CODE REVIEW

File: guides/source/active_record_validations.md [hunk 1]
```diff
@@ -1055,7 +1055,9 @@ person.errors[:name]
 
 ### `errors.add`
 
-The `add` method lets you manually add messages that are related to particular attributes. You can use the `errors.full_messages` or `errors.to_a` methods to view the messages in the form they might be displayed to a user. Those particular messages get the attribute name prepended (and capitalized). `add` receives the name of the attribute you want to add the message to, and the message itself.
+The `add` method lets you manually add messages that are related to particular attributes. It takes as arguments the name of the attribute receiving the message, and the message itself.
```
Comment (kaspth) [inline guides/source/active_record_validations.md]: I find the explanation for the first argument tough to understand as it's very abstract. Is there a way to make it more concrete?

File: guides/source/active_record_validations.md [hunk 2]
```diff
@@ -1055,7 +1055,9 @@ person.errors[:name]
...
+
+You can use the `errors.full_messages` or `errors.to_a` methods to view the messages in the form they might be displayed to users, with the capitalized attribute name prepended to the messages.
```
Comment (kaspth) [inline guides/source/active_record_validations.md]: If only they might (versus will), maybe it's not such valuable information to add here? :smile:

Comment (jonatack) [inline guides/source/active_record_validations.md]: Thanks for the review! Depends how you write your "display_error_messages" partial (using message or full_messages), right? :smiley: 
Do you prefer "will"?

File: guides/source/active_record_validations.md [hunk 3]
```diff
@@ -1073,12 +1075,12 @@ person.errors.full_messages
  # => ["Name cannot contain the characters !@#%*()_-+="]
 ```
 
-Another way to do this is using `[]=` setter
+An equivalent to `errors#add` is to use `<<` to append a message to the `errors.messages` array for an attribute:
```
Comment (jonatack) [inline guides/source/active_record_validations.md]: (Fixing lines 1076 and 1081 is the main reason for the PR).

File: guides/source/active_record_validations.md (hunk 1 above)
Comment (jonatack) [inline guides/source/active_record_validations.md]: How about "It takes as arguments the name of the attribute and the message."

File: guides/source/active_record_validations.md (hunk 2 above)
Comment (matthewd) [inline guides/source/active_record_validations.md]: > Do you prefer "will"?

No. "Might" is fine, IMO... but if it needs to be rephrased, something more along the lines of "in a form suitable to be displayed to users" is probably closer.

**Comment by jonatack** [comment]
Will squash.

**Comment by kaspth** [comment]
@jonatack Something seems to have gone wrong. There's now 22 commits here.

**Comment by jonatack** [comment]
Yes, I've messed up my git history and cannot rebase -i to squash. Closing.

**Comment by jonatack** [comment]
Here was the updated text:

The `add` method lets you add an error message related to a particular attribute. It takes as arguments the attribute and the error message.

The `errors.full_messages` method (or its equivalent, `errors.to_a`) returns the error messages in a user-friendly format, with the capitalized attribute

**Comment by kaspth** [comment]
Updated text looks great, :+1:

Are you filing a new issue or salvaging the branch by hard resetting to master and cherry-picking your commit?

> Den 24/03/2015 kl. 08.21 skrev Jon Atack <notifications@github.com>:
> 
> Here was the updated text:
> 
> The add method lets you add an error message related to a particular attribute. It takes as arguments the attribute and the error message.
> 
> The errors.full_messages method (or its equivalent, errors.to_a) returns the error messages in a user-friendly format, with the capitalized attribute
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/rails/rails/pull/19457#issuecomment-85373597>.
> 

--
Kasper

**Comment by jonatack** [comment]
Thanks.
Will try the latter, and if no go then the former.

**Pull request closed without merging**